test/unit/gateways/reach_test.rb

```diff
@@ -163,6 +163,28 @@ def test_stored_credential_with_no_store_credential_parameters
     end.respond_with(successful_purchase_response)
   end
 
+  def test_stored_credential_with_reach_payment_model
+    reach_options = [
+      { payment_model: 'CIT-Setup-Scheduled' },
+      { payment_model: 'CIT-Setup-Unscheduled-MIT' },
+      { payment_model: 'CIT-Setup-Unscheduled' },
+      { payment_model: 'CIT-Subsequent-Unscheduled' },
+      { payment_model: 'MIT-Subsequent-Scheduled' },
+      { payment_model: 'MIT-Subsequent-Unscheduled' }
+    ]
```

Comment:
Comment:

What is the point on having a hash in something that should be an array ?

something like:

```ruby
  reach_options = [
        'CIT-Setup-Scheduled',
        'CIT-Setup-Unscheduled-MIT',
        'CIT-Setup-Unscheduled',
        'CIT-Subsequent-Unscheduled',
        'MIT-Subsequent-Scheduled',
        'MIT-Subsequent-Unscheduled'
  ]

    reach_options.each do |payment_model|
      @options[:payment_model] = payment_model
```

Reply:
Oh yes, make sense, I will update it, thanks